checks/exec.go

```diff
@@ -224,9 +227,54 @@ func setupConnection(ctx *context.Context, check v1.ExecCheck, cmd *exec.Cmd) er
 	return nil
 }
 
-func checkCmd(ctx *context.Context, cmd *exec.Cmd, result *pkg.CheckResult) (results pkg.Results) {
+func checkCmd(ctx *context.Context, check v1.ExecCheck, cmd *exec.Cmd, result *pkg.CheckResult) (results pkg.Results) {
 	details := runCmd(ctx, cmd)
 	result.AddDetails(details)
+
+	for _, artifactConfig := range check.Artifacts {
+		switch artifactConfig.Path {
+		case "/dev/stdout":
+			result.Artifacts = append(result.Artifacts, pkg.ArtifactResult{
+				Connection:  DefaultArtifactConnection,
+				Content:     []byte(details.Stdout),
+				ContentType: "text/plain",
+				Path:        "stdout",
+			})
+
+		case "/dev/stderr":
+			result.Artifacts = append(result.Artifacts, pkg.ArtifactResult{
+				Connection:  DefaultArtifactConnection,
+				Content:     []byte(details.Stderr),
+				ContentType: "text/plain",
+				Path:        "stderr",
+			})
+
+		default:
+			paths := utils.UnfoldGlobs(artifactConfig.Path)
+			for _, path := range paths {
+				artifact := pkg.ArtifactResult{
+					Connection: DefaultArtifactConnection,
+				}
+
+				file, err := os.Open(path)
+				if err != nil {
+					logger.Errorf("error opening file. path=%s; %w", path, err)
+					continue
+				}
+				defer file.Close()
+
+				if artifact.Content, err = io.ReadAll(file); err != nil {
+					logger.Errorf("error reading file. path=%s; %w", path, err)
+					continue
+				}
+
+				artifact.Path = path
+				artifact.ContentType = http.DetectContentType(artifact.Content)
```

Comment:
I think we can just use a https://github.com/nanmu42/limitio LimitedWriter and the https://github.com/gabriel-vasile/mimetype library for this so that we buffer at most 512KB to feed into mimetype detection, for larger files we wont necessarily detect the mime type.